src/test/test.ml

```diff
@@ -21,4 +21,92 @@ module Basic = struct
   let tests = [ ("shark blocks", `Quick, test_shark_block) ]
 end
 
-let () = Alcotest.run "shark" [ ("basic", Basic.tests) ]
+module CommandParsing = struct
```

Comment:
Thanks for adding the tests! I think it would be more concise to define a pretty-printing function for commands and to use some more Alcotest combinators to remove the explicit calls to `Alcotest.fail`.

I started having a go, I can do the rest if you think the idea is ok?


```ocaml
  let command = Alcotest.of_pp Shark.Command.pp

  let test_python_command_module () =
    let testcase = "python3 -m some.module.code arg1 arg2" in
    let expected =
      Shark.Command.v ~name:"some.module.code"
        ~args:[ "-m"; "some.module.code"; "arg1"; "arg2" ]
        ~file_args:[]
    in
    let test = Shark.Command.of_string testcase in
    Alcotest.(check (option command)) "Command" test (Some expected)
```

with command.ml becoming:

```diff
diff --git a/src/lib/command.ml b/src/lib/command.ml
index 297926e2..7388c8d6 100644
--- a/src/lib/command.ml
+++ b/src/lib/command.ml
@@ -1,6 +1,11 @@
 open Astring
+open Sexplib.Conv
 
 type t = { name : string; args : string list; file_args : string list }
+[@@deriving sexp]
+
+let pp ppf t = Sexplib.Sexp.pp_hum ppf (sexp_of_t t)
+let v ~name ~args ~file_args = { name; args; file_args }
```

Reply:
Sure, go for it - unfortunately I won't have time to do that myself.